xtevent/_eventivstatic.ado

```diff
@@ -1,172 +1,184 @@
-		loc cmd "ivregress 2sls"
-		loc ffe ""
-		}
-		else {
-			loc cmd "xtivreg"
-			loc ffe "fe"
-		}
-		`cmd' `varlist' (`proxy' = `insvars') `z' `tte' [`weight'`exp'] if `touse' , `ffe' `options'
-	}
-	else {
-		loc cmd "reghdfe"
-		if "`fe'" == "nofe" & "`te'"=="" & "`absorb'"=="" {						
-			loc noabsorb "noabsorb"
-			loc abs ""
-		}
-		else if "`fe'" == "nofe" & "`te'"=="" & "`absorb'"!="" {						
-			loc noabsorb "noabsorb"
-			loc abs "absorb(`absorb')"
-		}
-		else if "`fe'" == "nofe" & "`te'"!=""{
-			loc abs "absorb(`te' `absorb')"	
-			loc noabsorb ""
-		}
-		else if "`fe'" != "nofe" & "`te'"==""{
-			loc abs "absorb(`i' `absorb')"	
-			loc noabsorb ""
-		}
-		else {
-			loc abs "absorb(`i' `te' `absorb')"	
-			loc noabsorb ""
-		}
-		ivreghdfe `varlist' (`proxy' = `insvars') `z' [`weight'`exp'] if `touse', `absorb' `noabsorb' `options'
-	}
-	
-	
-	
-	
-	if "`te'" == "note" loc tte ""
-	else loc tte "i.`t'"
-	
-	mat `bb' = e(b)
-	mat `VV' = e(V)
-	mat `delta'=e(b)
-	mat `Vdelta'=e(V)	
-	
-	* Drop variables
-	
-	if "`instype'"=="numlist" {
-		foreach v in `proxyiv' {
-			drop _f`v'`z' 
-		}
-	}
-	
-	tokenize `varlist'
-	loc depvar "`1'"
-	
-	return matrix b = `bb'
-	return matrix V = `VV'
-	return matrix delta = `delta'
-	return matrix Vdelta = `Vdelta'
-	loc names: subinstr global names ".." " ", all
-	loc names: subinstr local names `"""' "", all
-	return local names = "`names'"	
-	return local cmd = "`cmd'"
-	return local depvar = "`depvar'"
-		
-end
+* _eventivstatic.ado 1.00 Aug 24 2021
+
+version 11.2
+
+cap program drop _eventivstatic
+program define _eventivstatic, rclass
+	#d;
+	syntax varlist(fv ts numeric) [aw fw pw] [if] [in], /* Covariates go in varlist. Can add fv ts later */
+	panelvar(varname) /* Panel variable */
+	timevar(varname) /* Time variable */
+	policyvar(varname) /* Policy variable */	
+	proxy (varlist numeric) /* Proxy variable(s) */
+	[
+	proxyiv(string) /* Instruments. Either numlist with lags or varlist with names of instrumental variables */
+	nofe /* No fixed effects */
+	note /* No time effects */	
+	reghdfe /* Use reghdfe for estimation */
+	absorb(string) /* Absorb additional variables in reghdfe */ 
+	impute(string)
+	STatic
+	*
+	]
+	;
+	#d cr
+	
+	marksample touse
+	
+	tempvar kg
+	* kg grouped event time, grouping outside window
+	
+	tempname delta Vdelta bb VV bb2 VV2 delta2 Vdelta2 deltay Vdeltay deltax Vdeltax deltaxsc bby bbx VVy VVx 
+	* bb delta coefficients
+	* VV variance of delta coefficients
+	* bb2 delta coefficients for overlay plot
+	* VV2 variance of delta coefficients for overlay plot
+	* delta2 included cefficientes in overlaty plot
+	* VVdelta2 variance of included delta coefficients in overlay plot
+	
+	loc i = "`panelvar'"
+	loc t = "`timevar'"
+	loc z = "`policyvar'"
+	
+	loc leads : word count `proxy'
+	if "`proxyiv'"=="" & `leads'==1 loc proxyiv "select"
+	
+	* If proxy specified but no proxyiv, assume numlist for leads of policyvar
+	if "`proxyiv'"=="" {

```

Comment:
Will an unspecified proxyiv always be captured with `""`? Wondering if more conditions are necessary to account for different ways proxyiv could be unspecified, such as if proxyiv is missing?

Reply:
Thanks, I'm checking if "" is sufficient to capture an unspecified proxyiv.

Reply:
If `proxyiv` is unspecified then its local should be missing and both sides of the equality will be equal to `""', so I think this is covered.